public/docs/_examples/animations/ts/src/hero-list-auto.component.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Component,
+  Input,
+  animation,
+  state,
+  style,
+  animate,
+  transition,
+  group
+} from '@angular/core';
+import {Hero, Heroes} from './hero.service';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-list-auto',
+  // #docregion template
+  template: `
+    <div class="hero"
+         *ngFor="let hero of heroes.get()"
+         @shrinkOut="'in'">
+      {{hero.name}}
+    </div>
+  `,
+  // #enddocregion template
+  styleUrls: ['hero-list.component.css'],
+  // #docregion animationdef
+  animations: [
+    animation('shrinkOut', [
+      state('in', style({height: '*'})),
+      transition('* => void', [
+        style({height: '*'}),
+        animate(250, style({height: 0}))
+      ])
+    ])
```

Comment:
@matsko Should we also be able to do a similar thing for entering? I tried it but couldn't figure out how to make it so that it has a usable computed height in `void => *`